### ISSUE TITLE
BeanDeserializerFactory.constructSettableProperty does not set _valueTypeDeserializer nor _valueDeserializer for Jdk types

### ISSUE THREAD

**Issue by ink-ua** [FasterXML/jackson-databind#233]
Consider we have an annotated bean with the field of some JDK type, e.g. String. Using the following factory method we try to create deserializer inside our own BeanDeserializerModifier:
BeanDeserializerFactory.instance.createBeanDeserializer(ctxt, ctxt.constructType(result.getClass()), desc);

Somewhere inside there is an invocation of BeanDeserializerFactory.constructSettableProperty which contains the following code:

JsonDeserializer<Object> propDeser = findDeserializerFromAnnotation(ctxt, mutator);
// some unrelated code here
if (propDeser != null) {
     prop = prop.withValueDeserializer(propDeser);
}

since our bean type is not annotated at all findDeserializerFromAnnotation will return null and ValueDeserializer remains unset (jackson 2.0, in the latest version MISSING_VALUE_DESERIALIZER value) so when we try to deserialize this property we get NPE (or I belive an exception in the newer jackson version) in SettableBeanProperty.deserialize:

if (_valueTypeDeserializer != null) {
            return _valueDeserializer.deserializeWithType(jp, ctxt, _valueTypeDeserializer);
}
return _valueDeserializer.deserialize(jp, ctxt);

So basically String can't be deserialized into String, which doesn't seem right. If there are no annotations but the type is standard it should be possible to have deserializer out of the box. I guess we could reuse BeanDeserializerFactory.findStdDeserializer (or, at least a part of it) if findDeserializerFromAnnotation returns null:

Class<?> cls = type.getRawClass();
        // note: we do NOT check for custom deserializers here; that's for sub-class to do
        JsonDeserializer<Object> deser = _simpleDeserializers.get(new ClassKey(cls));
        if (deser != null) {
            return deser;
        }
